event_handling/include/essentials/NotifyTimer.hpp

```diff
@@ -0,0 +1,180 @@
+#pragma once
+
+#include "ITrigger.hpp"
+
+#include <chrono>
+#include <condition_variable>
+#include <iostream>
+#include <thread>
+#include <vector>
+
+namespace essentials
+{
+
+template <class NotificationClass>
+using t_notificationcallback = void (NotificationClass::*)();
+
+/**
+ * The NotifyTimer allows to have a callback-member function be called in a fixed
+ * interval. The calling is delayed after each start by the given delay.
+ * @tparam NotificationClass
+ */
+template <class NotificationClass>
+class NotifyTimer : public ITrigger
+{
+public:
+    NotifyTimer(std::chrono::milliseconds msInterval, std::chrono::milliseconds msDelayedStart, t_notificationcallback<NotificationClass> callback,
+            NotificationClass* obj);
```

Comment:
don't see any advantage of making template class if we are anyways going to take callback and object in arguments.

Reply:
so it will work without being a template class?